api/app/Policies/UserPolicy.php

```diff
@@ -118,4 +181,57 @@ protected function applicantHasAppliedToPoolInTeams(User $applicant, $teamIds)
             })
             ->exists();
     }
+
+    /*******************  ROLE CHECKING  *******************/
+
+    /**
+     * Function to check if the acting user can update a team based role
+     *
+     * @return bool
+     */
+    protected function teamAbleToCheck(User $actor, string $roleId, string $teamId)
+    {
+        $role = Role::findOrFail($roleId);
+        $team = Team::findOrFail($teamId);
+
+        switch ($role->name) {
+            case 'pool_operator':
+                return $actor->isAbleTo('assign-any-teamRole');
+            case 'process_operator':
+                return $actor->isAbleTo('update-any-processOperatorMembership') || $actor->isAbleTo('update-team-processOperatorMembership', $team);
```

Comment:
This one is a bit confusing, but will need some extra logic after #10364. Might be good to put it in now but leave it commented out.
```
case 'process_operator':
                // Community roles have the update-team-processOperatorMembership permission, and it should give them the ability to assign processOperator roles to pools in their community.
               // TODO: uncomment this in issue #10364
                // $pool = $team->teamable; // If we're adding a processOperator to a team, that team should represent a pool. Need validation?
                // $community = $pool->community;
                return $actor->isAbleTo('update-any-processOperatorMembership')
                    || $actor->isAbleTo('update-team-processOperatorMembership', $team)
                    \\ || $actor->isAbleTo('update-team-processOperatorMembership', $community->team)
                    ;
```

Reply:
Yeah okay, I added that in with https://github.com/GCTC-NTGC/gc-digital-talent/pull/10791/commits/ef143eacc5132ff8a783dacdd25cf31218e93713
I also edited the AC of the this issue https://github.com/GCTC-NTGC/gc-digital-talent/issues/10364 to look for TODOs